class Solution {
    public void setZeroes(int[][] matrix) {
        boolean firstRow = false;
        boolean firstColumn = false;
        for(int i = 0 ; i < matrix.length ; i++){
            for(int j = 0 ; j < matrix[0].length ; j++){
                if(matrix[i][j] == 0){
                    if(i == 0){
                        firstRow = true;
                    }
                    if(j == 0){
                        firstColumn = true;
                    }
                    matrix[0][j] = 0;
                    matrix[i][0] = 0;
                }
            }
        }
                for(int i = matrix.length-1  ; i >= 1 ; i--){
                    for(int j = matrix[0].length-1 ; j >= 1 ; j--){
                        if(matrix[i][0] == 0 || matrix[0][j] == 0){
                            matrix[i][j] = 0;
                        }
                    }
                }
                        if(firstRow){
                            for(int j = 0 ; j < matrix[0].length ; j++){
                                matrix[0][j] = 0;
                            }
                        }
                        if(firstColumn){
                            for(int i = 0 ; i < matrix.length ; i++){
                                matrix[i][0] = 0;
                            }
                        }
    }
}